src/Host/Broker/Impl/Sessions/Session.cs

```diff
@@ -160,6 +161,18 @@ public class Session {
             HostToClientWorker(_process.StandardOutput.BaseStream, hostEnd).DoNotWait();
         }
 
+        private void StartSession() {
+            _process.Start();
+            _process.WaitForExit(250);
+            if (_process.HasExited && _process.ExitCode != 0) {
+                var message = ErrorCodeConverter.MessageFromErrorCode(_process.ExitCode);
```

Comment:
User errors should always be positive. By the example you give, it sounds like OS errors would always be negative? If so, perhaps we could use that to differentiate?